Update HttpStatusPushBase to accept wantXXX kwargs

rayalan opened the pull request:
- Previous, HttpStatusPush nominally accepted them, but also passed them to
  HttpStatusPushBase which did not accept them. However, the base is the class
  responsible for storing the kwargs, so now it configures them as well.

----
I don't see any reason for HttpStatuPushBase to not also configure the wantXXX kwargs since it is the one with the 'neededDetails' member and the one using it.

rayalan commented:
(I expect unittest failures due to master branch problems with 2.6; I'll rebase when fixes to those get committed.)

tardyp commented:
@rayalan thanks for those fixes on http reporters!

You can rebase, I fixed the master branch for 2.6 support (was due to external dep update)

tardyp commented:
actually, the tests are failing due to this code. please check..
```

===============================================================================
[FAIL]
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/twisted/internet/defer.py", line 1128, in _inlineCallbacks
    result = g.send(result)
  File "/home/travis/build/buildbot/buildbot/master/buildbot/test/unit/test_reporter_http.py", line 118, in test_wantKwargsCheck
    BuildLookAlike(), auth=('username', 'passwd'))])
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/twisted/trial/_synctest.py", line 437, in assertEqual
    super(_Assertions, self).assertEqual(first, second, msg)
  File "/opt/python/2.7.9/lib/python2.7/unittest/case.py", line 513, in assertEqual
    assertion_func(first, second, msg=msg)
  File "/opt/python/2.7.9/lib/python2.7/unittest/case.py", line 506, in _baseAssertEqual
    raise self.failureException(msg)
twisted.trial.unittest.FailTest: [] != [call(u'serv', { any build }, auth=('username', 'passwd'))]
buildbot.test.unit.test_reporter_http.TestHttpStatusPush.test_wantKwargsCheck
-------------------------------------------------------------------------------
```

rayalan commented:
Alright...
:white_check_mark: Tests fixed. 
:white_check_mark: Rebased against latest master. 

But now AppVeyorBot seems to be giving me grief about the Windows build. Do I need to be paying attention to it too? The failure doesn't seem connected to the code I changed.

rayalan commented:
@tardyp Ready for review again I think (not sure if your involvement on this request meant all notifications got sent your way or not).

tardyp commented:
thank you!
